ledger_device_sdk/src/nbgl.rs

```diff
@@ -227,49 +163,33 @@ impl<'a> NbglHomeAndSettings<'a> {
         version: &'a str,
         author: &'a str,
     ) -> NbglHomeAndSettings<'a> {
+        let mut v: Vec<CString> = Vec::new();
+        v.push(CString::new(app_name).unwrap());
+        v.push(CString::new(version).unwrap());
+        v.push(CString::new(author).unwrap());
         NbglHomeAndSettings {
-            info_contents: [
-                self.c_string_helper.to_cstring(app_name).unwrap().as_ptr() as *const c_char,
-                self.c_string_helper.to_cstring(version).unwrap().as_ptr() as *const c_char,
-                self.c_string_helper.to_cstring(author).unwrap().as_ptr() as *const c_char,
-            ],
+            info_contents: v,
             ..self
         }
     }
 
     pub fn settings(
         self,
         nvm_data: &'a mut AtomicStorage<[u8; SETTINGS_SIZE]>,
-        settings_strings: &[[&str; 2]],
+        settings_strings: &[[&'a str; 2]],
     ) -> NbglHomeAndSettings<'a> {
         unsafe {
             NVM_REF = Some(transmute(nvm_data));
         }
 
-        if settings_strings.len() > SETTINGS_SIZE {
```

Comment:
I think we should put a restriction on how many settings_strings are allowed, it will lead to an app crash at line 220 if there are more `setting_strings` item than there are slots in `SWITCH_ARRAY`

Reply:
Maybe break the loop at line 220 if we reach max number of items in SWITCH_ARRAY, preventing a crash

Reply:
True, I have re-enabled the settings length check in the function to prevent any further crash.